### ISSUE TITLE
X-Men 2: Clone Wars - No Audio on first boot

### ISSUE THREAD

**Issue by R9delta** [MiSTer-devel/Genesis_MiSTer#167]
X-Men 2 - Clone Wars (USA, Europe).gen
CRC32: 710BC628

Hi, i'm having an issue where I have no audio output with this game until I reset it. This is easily reproduced (at least for me) and happens with 20210105 however 20201105 and older do not exhibit this.

**Comment by SegaSnatcher** [comment]
I hear audio on first boot.  No issues here.

**Comment by rafaelzig** [comment]
Also happens on my setup, I have to reset to have the audio back to normal.

**Comment by birdybro** [comment]
Works fine on mine. I'm playing over HDMI, and with latest core as of today. It's possible this is related to the signal chain of your audio from HDMI, this happens with some receivers and systems. The game starts up and sounds like static probably, and your system is blocking it out. If you are going through a receiver, try bypassing the receiver and see if that changes anything. If you are going through an hdmi switch, try bypassing that. Try disabling the hdmi 96k audio option in the ini settings on the mister. Etc... It could be any number of things. But all's I know is, it's working on mine.

**Comment by birdybro** [comment]
@R9delta does this still happen for you on latest core and latest main framework+linux?

If so, what is your configuration of your MiSTer? Is it over HDMI, over analog i/o board, where is the sound being generated? etc... I tested this again today while reviewing all for the open issues and I noticed this still works fine for me over HDMI. I also tested with DDR3 set to be sure this wasn't a bug for DDR3, and I still got sound.

**Comment by Kopert** [comment]
@birdybro I just ran into this issue today when I first tried loading this game, then I came here looking if it was a known problem. Running release 20211126, rom MD5 b687660446f6e49dfd465cd0ef9f0bc3. Tried running a beta rom of the same game, dated 1994-12-16, the problem is the same.

As soon as I reset the Genesis core, it works again. If I reboot the MiSTer, it doesn't work when the game is first started; if I load another game and come back to it, it doesn't work either. It affects both the VGA->YPbPr output on the analog board (connected to a consumer CRT TV) and on the HDMI output (connected to a modern a/v receiver). I have the sdram module installed.

Haven't seen this occur with any other game, but I haven't tried many.

I'll play around with settings to see if I can pinpoint something else, will report back if I find anything new. Let me know if I can provide any useful information or testing.

**Comment by Kopert** [comment]
Hey, I just found something interesting. If I set `ROM Storage` to `DDR3` instead of `Auto`, the problem does not happen!

My SDRam module is a 2.9 128mb from UltimateMister, it works great, never had any problems on CPS 1.5 or NeoGeo games.

So maybe the time it takes to load the rom into storage is the issue, since resetting the game does not involve reloading it?

**Comment by birdybro** [comment]
> @birdybro I just ran into this issue today when I first tried loading this game, then I came here looking if it was a known problem. Running release 20211126, rom MD5 b687660446f6e49dfd465cd0ef9f0bc3. Tried running a beta rom of the same game, dated 1994-12-16, the problem is the same.
> 
> As soon as I reset the Genesis core, it works again. If I reboot the MiSTer, it doesn't work when the game is first started; if I load another game and come back to it, it doesn't work either. It affects both the VGA->YPbPr output on the analog board (connected to a consumer CRT TV) and on the HDMI output (connected to a modern a/v receiver). I have the sdram module installed.
> 
> Haven't seen this occur with any other game, but I haven't tried many.
> 
> I'll play around with settings to see if I can pinpoint something else, will report back if I find anything new. Let me know if I can provide any useful information or testing.

Thanks for following up!

1. Is the region set correctly? If not make sure you toggle the region detection to "header" and save settings. Even if it's on "header" already, toggle it a couple times and save settings again. The game should boot and the region should show "US" if it's working right.
2. In your analog video vs your hdmi video setups, are you using different audio output for each? Just want to see how I can test this on my end to recreate what you are doing. It might also help rule out things like CEC causing issues with HDMI audio if you are using a audio splitter at some point, etc...

**Comment by Kopert** [comment]
Thanks for looking into this.

>     1. Is the region set correctly?

The region was set correctly to US with automatic header detection. I forced it to US with no header detection and there was no change. I also tried to force it to EU for testing purposes and apparently the issue does not happen if it is set to that region. Once I set it back to US, the problem returned.

>     2. In your analog video vs your hdmi video setups, are you using different audio output for each?

Yes. My analog audio is coming out of the analog i/o board using a 3.5mm to RCA cable directly to the TV. The HDMI output is connected to an a/v receiver with CEC support. I have tested it with no HDMI cable connected and there was no change.

**Comment by birdybro** [comment]
Really strange, I am not having this issue still, the sound comes through perfectly for me on my setup. The only thing I could imagine now is your Main MiSTer binary file is corrupt, your linux image is corrupt, or something is wrong with your hardware. I know that's probably not a very satisfactory answer and could probably feel frustrating, but can you try a fresh MiSTer install on another SD Card, update it fully with the new downloader script (do it twice to make sure it's all up to date), and then test again?

**Comment by Kopert** [comment]
**TL;DR - How to fix the issue: delete the config/Genesis.CFG file**

@birdybro thanks for your efforts and don't worry about providing a "frustrating answer", as a fellow dev I know there's not much you can do if you can't reproduce the issue.

Before I even finished reading your message, I remembered that I had another SD card that came with my MiSTer (I got it prebuilt) and I figured it would be a quick test. It was, and on that sd card it worked. So I decided to try each part of that installation until I figured out what is wrong.

- Tried the core .rbf file, no changes
- Tried the .ini file, no changes
- Tried the MiSTer file, no changes

What fixed the issue was deleting the config/Genesis.CFG file from the sd card. [Here's the broken .cfg file](https://github.com/MiSTer-devel/Genesis_MiSTer/files/7729088/Genesis.zip) if you wish to test it yourself.

I know this sounds like I had something wrong set up, but I checked the settings and couldn't find any differences - I even made a few changes to the settings to match what I had previously and it still works.

The issue is fixed for me, but I feel it might be worth it to try and figure out what the issue is. For reference, the files are:

`60 00 00 00 00 00 00 00` - Broken Genesis.cfg
`40 20 00 00 00 00 00 00` - Working Genesis.cfg

Does this help?

**Comment by birdybro** [comment]
Yeah, typically when there are changes to the OSD options that are significant, you should toggle a setting and click "save settings" which will overwrite the config. That is weird, but I've seen weird stuff happen which was solved by re-saving settings before.

**Comment by Kopert** [comment]
Just re-saving settings didn't fix it this time, though, because I'm positive I did that when testing different regions.

Anyway, unless you wish to investigate this further, I'd consider this issue fixed on my end. Feel free to let me know if you'd like me to test anything else.

**Comment by birdybro** [comment]
I'd consider it a resolved issue because it seems like a broken cfg file was to blame :P
